lib/pyvirtualbench.py

```diff
@@ -22,11 +22,15 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
+import struct
 from ctypes import c_bool, c_size_t, c_double, c_uint8, c_uint16, c_int32, c_uint32, c_int64, c_uint64, c_char_p, c_wchar, c_wchar_p, Structure, c_int, cdll, byref
 from enum import IntEnum
 
 NIVB_LIBRARY_VERSION = 302039040 # 18.0.0f0, is found in nivirtualbench.h
 
+PYTHON_ARCHITECTURE = 8 * struct.calcsize("P")  # 32 or 64 bit
+
+

```

Comment:
The Python docs recommend using `sys.maxsize > 2**32` for 64 bit detection (see [here](https://docs.python.org/3.8/library/platform.html)). I would also recommend storing the correct function to use once:
```
if sys.maxsize > 2**32:
    c_sysint = c_int64
else:
    c_sysint = c_int
```
later:
```
self.instrument_handle = c_sysint(0)
```

Reply:
thats the more elegant solution I was asking for 😄
Apparently I was too exicted that it works at all to think this through.

The 64 bit detection I got from stackoverflow, since `sys.maxsize` is only compatible with Python 2.6 and later. But since the readme states that python >3.4 is required, this shouldn't be an issue. Changed it like you suggested.

Thanks for your suggestions!

Reply:
If you want to make this even nicer, don't bother with the calculation at all.  On the long import line from ctypes, also import c_void_p.  This is the actual type of a handle anyway, and then all the locations that currently use "c_int" for a handle simply change to "c_void_p".

Reply:
I've gone ahead and accepted the Python recommended way ```sys.maxsize > 2**32```.  Thanks everyone for the great discussion and suggestions.